crates/evm/src/evm/handler.rs

```diff
@@ -401,8 +405,16 @@ impl<SPEC: Spec, EXT: CitreaExternalExt, DB: Database> CitreaHandler<SPEC, EXT,
         context: &mut Context<EXT, DB>,
         result: FrameResult,
     ) -> Result<ResultAndState, EVMError<<DB as Database>::Error>> {
-        let diff_size =
-            calc_diff_size::<EXT, SPEC, DB>(context).map_err(EVMError::Database)? as u64;
+        let uncompressed_size =
+            calc_diff_size::<EXT, SPEC, DB>(context).map_err(EVMError::Database)?;
+
+        let diff_size = if SPEC::enabled(SpecId::CANCUN) {
```

Comment:
Well, better avoid muts if they can be avoided.

```
let uncompressed_size = ...
let compression_ratio = if SPEC::enabled(SpecId::CANCUN) {
    BROTLI_COMPRESSION_PERCENTAGE / 100
} else {
    1.0
}
let diff_size = (uncompressed_size * compression_ratio) as u64;
```

Reply:
I wanted to avoid using floats and muts at the same time, that's why I used percentage like other discount constants. If using floats is fine I can go with Roman's suggestion?

Reply:
made changes